pkg/pool-hooks/contracts/StableSurgeHook.sol

```diff
@@ -187,7 +191,100 @@ contract StableSurgeHook is BaseHooks, VaultGuard, SingletonAuthentication {
         address pool,
         uint256 staticSwapFeePercentage
     ) public view override onlyVault returns (bool, uint256) {
-        return (true, getSurgeFeePercentage(params, pool, staticSwapFeePercentage));
+        return (true, getSurgeSwapFeePercentage(params, pool, staticSwapFeePercentage));
+    }
+
+    function getSurgeSwapFeePercentage(PoolSwapParams calldata params, address pool, uint256 staticSwapFeePercentage) public view returns (uint256) {
+        uint256 numTokens = params.balancesScaled18.length;
+
+        uint256 amountCalculatedScaled18 = StablePool(pool).onSwap(params);
+
+        uint256[] memory newBalances = new uint256[](numTokens);
+        ScalingHelpers.copyToArray(params.balancesScaled18, newBalances);
+
+        if (params.kind == SwapKind.EXACT_IN) {
+            newBalances[params.indexIn] += params.amountGivenScaled18;
+            newBalances[params.indexOut] -= amountCalculatedScaled18;
+        } else {
+            newBalances[params.indexIn] += amountCalculatedScaled18;
+            newBalances[params.indexOut] -= params.amountGivenScaled18;
+        }
+
+        return _getSurgeFeePercentage(params, pool, staticSwapFeePercentage, newBalances);
+    }
+
+    /// @inheritdoc IHooks
+    function onAfterAddLiquidity(
+        address,
+        address pool,
+        AddLiquidityKind kind,
+        uint256[] memory amountsInScaled18,
+        uint256[] memory amountsInRaw,
+        uint256,
+        uint256[] memory balancesScaled18,
+        bytes memory
+    ) public view override returns (bool success, uint256[] memory hookAdjustedAmountsInRaw) {
+        hookAdjustedAmountsInRaw = amountsInRaw; // don't adjust amounts in.
```

Comment:
Might get optimized by the compiler, but this might be copying twice (here and in the return). Other hooks don't have this assignment (or the named return value at all), and just return `amountsInRaw`. Every path does a return here, so not declaring `hookAdjustedAmountsInRaw` and just returning `amountsInRaw` everywhere seems cleaner.